src/ui_utils.c

```diff
@@ -3230,14 +3230,22 @@ gboolean ui_encodings_combo_box_set_active_encoding(GtkComboBox *combo, gint enc
 	return FALSE;
 }
 
-void ui_menu_popup(GtkMenu* menu, GtkMenuPositionFunc func, gpointer data, guint button, guint32 activate_time)
+
+/* opens menu at caret position (if ScintillaObject is passed) or at current pointer position */
+void ui_menu_popup(GtkMenu* menu, ScintillaObject *sci)
 {
-	/* Use appropriate function for menu popup:
-		- gtk_menu_popup_at_pointer is not available on GTK older than 3.22
-		- gtk_menu_popup is deprecated and causes issues on multimonitor wayland setups */
-#if GTK_CHECK_VERSION(3,22,0)
-	gtk_menu_popup_at_pointer(GTK_MENU(menu), NULL);
-#else
-	gtk_menu_popup(GTK_MENU(menu), NULL, NULL, func, data, button, activate_time);
-#endif
+	if (!sci)
+		gtk_menu_popup_at_pointer(GTK_MENU(menu), NULL);
+	else
```

Comment:
Since we need this special case only for the tag completion popup, we might also keep the special logic in smbols.c as before and then we can drop this helper function altogether and use `gtk_menu_popup_at_pointer()` directly.

What do you think?

Reply:
You're right. This could be a lot simpler now, that we use only reasonably modern GTK. Actually the only reason why this helper function was created was to hide that ugly `#ifdef GTK_CHECK_VERSION(3, 22, 0)` in all the places that show menu.